### ISSUE TITLE
Persisting Pagination

### ISSUE THREAD

**Issue by cowan14** [aries-auto/ariesautomotive#22]
Alex....you know what I mean. right?

**Comment by DavidVaini** [comment]
To clarify when you are on the category parts listing page and you click "view more" to load more parts, it doesnt keep track of paging in the url. So when he goes to a part, and then hits the back button it goes to the first page of parts on the category.

**Comment by ninnemana** [comment]
Hard part of implementing this is back tracking the pagination. Current pagination model appends parts to the display, so if I back to page 3 of category parts, how do pages 1 & 2 get injected?

**Comment by DavidVaini** [comment]
my guess is it would have to take the url param so 3 for page 3. Then it would have to loop from 1-3 and append.

So if you do page 1, it would loop only the first page and if you were on the 6th page it would loop through to the 6th one and then focus to the 6th page.

**Comment by mickelsonm** [comment]
I think a good solution to this problem would be to implement a Stack (LIFO - to be specific) for navigation. As you navigate, it would add whatever data to the stack. Once you hit the back button, it would pop the stack and use that data.

TLDR: It would be front-end magic of implementing custom pagination logic.

**Comment by ninnemana** [comment]
The thing I worry about with your solution @DavidVaini is a janky user experience when it's injecting and re-focusing.

**Comment by DavidVaini** [comment]
Yeah, this issue is open for anyone to fix, so if my idea doesn't work, feel free to take a stab at it. I just wanted to clarify on it since Chris showed it to me earlier today.
